plugins/out_opensearch/opensearch.c

```diff
@@ -445,7 +462,41 @@ static int opensearch_format(struct flb_config *config,
         msgpack_pack_str_body(&tmp_pck, time_formatted, s);
 
         index = ctx->index;
-        if (ctx->logstash_format == FLB_TRUE) {
+        if (ctx->data_stream_mode == FLB_TRUE) {
+            index = ctx->data_stream_name;
+            if (ctx->ra_data_stream) {
+                if (ra_index != NULL) {
+                    flb_sds_destroy(ra_index);
+                }
+                /* a record accessor pattern exists for the index */
+                ra_index = flb_ra_translate(ctx->ra_data_stream,
+                                            (char *) tag, tag_len,
+                                            map, NULL);
+                if (!ra_index) {
+                    flb_plg_warn(ctx->ins, "invalid data stream translation from record accessor pattern, default to static index");
+                }
+                else {
+                    index = ra_index;
+                }
+
+                if (ctx->suppress_type_name) {
```

Comment:
It seems like this block is duplicated and could be extracted into a small static function, could you?